### PR TITLE
Created BOM, refactor folder structure

### PR DESCRIPTION
**Pull request by dpwspoon**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -35,52 +34,53 @@
...
-        <module>transport.http</module>
-        <module>transport.nio</module>
-        <module>transport.pipe</module>
-        <module>transport.sse</module>
-        <module>transport.ssl</module>
-        <module>transport.ws</module>
-        <module>transport.wseb</module>
-        <module>transport.wsn</module>
-        <module>transport.wsr</module>
+        <module>service/spi</module>
```
Comment (jitsni) [inline pom.xml]: I am ok with this way of organizing things under transport, service, resource.address. Other way is organize under protocols (for e.g: http, http/resource.address, http/transport, http/service.http.balancer, http/service.http.proxy etc). I wonder whether we thought about that one.

File: service/http.proxy/pom.xml [hunk 2]
```diff
@@ -7,6 +6,7 @@
         <groupId>org.kaazing</groupId>
         <artifactId>gateway</artifactId>
         <version>develop-SNAPSHOT</version>
+        <relativePath>../../pom.xml</relativePath>
```
Comment (jitsni) [inline service/http.proxy/pom.xml]: So, it doesn't reference parent, i.e. service/pom.xml ? Do we even have that one ?

**Comment by jitsni** [comment]
travisci build is failing with 
testTaskIsStartedWithProperConfiguration(org.kaazing.gateway.service.update.check.UpdateCheckServiceTest)  Time elapsed: 0.005 sec  <<< FAILURE!
java.lang.AssertionError: There is a jar in ./src/test/resources/gateway.server-5.0.0.8.jar that needs to be on the class path, fix your test runner settings

**Comment by dpwspoon** [comment]
The jar got excluded because of the .gitignore and a file move.  I've added it back

File: pom.xml (hunk 1 above)
Comment (dpwspoon) [inline pom.xml]: I'm up for refactoring it latter

**Pull request closed without merging**